File: L950-RevealCardsInIncreasingOrder.java
// https://leetcode.com/problems/reveal-cards-in-increasing-order/
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Queue<Integer> deckCards = new LinkedList<Integer>();
        int length = deck.length, i = 0, idx= 0;
        int[] res = new int[length];
        for (i = 0; i < length; i++) {
            deckCards.add(i);
        }
        i = 0;
        Arrays.sort(deck);
        while (!deckCards.isEmpty()) {
            i = deckCards.poll();
            res[i] = deck[idx++];
            if (!deckCards.isEmpty()) {
                deckCards.add(deckCards.poll());
            }
        }
        return res;
    }
}

File: L950-reveal_cards_in_increasing_order.py
# https://leetcode.com/problems/reveal-cards-in-increasing-order/
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        length = len(deck)
        que = [i for i in range(length)]
        res = [0] * length
        idx = 0
        deck.sort()
        while (que):
            i = que.pop(0)
            res[i] = deck[idx]
            idx += 1
            if (que):
                que.append(que.pop(0))
        return res
